lib/amdResolve.js

```diff
@@ -90,10 +91,20 @@ module.exports = function resolve(pattern, {
     const fileTypePattern  = new RegExp(`\\.${fileExtension}$`);
     const extensionPattern = new RegExp(`^${targetExtension}/`);
 
+    //check if the pattern has an alias
+    const matchinAliases =  Object.entries(aliases)
+        .filter( ([alias]) => new RegExp(`^${alias}`).test(pattern) );
```

Comment:
- Can alias be a Regexp?
- Can multiple alias be applied to a pattern?
- Your resolver won't work on the following pattern `ui/queueButton` and alias `'ui/queue': 'src'`, because it will do something like this: `srcButton`

Use mine, if you want from `tao-core-ui`

```
const resolveAlias = id => {
    for (let alias in aliases) {
        if (aliases.hasOwnProperty(alias)) {
            let afterAlias;
            if (id.indexOf(alias) === 0 && (afterAlias = id.substring(alias.length))[0] === '/') {
                return `${aliases[alias]}${afterAlias}`;
            }
        }
    }
    return id;
};
```

Reply:
A test example that proves:

|- extA
    |- coreButton
        |- a.js

```
 it("should resolve a pattern with an complex aliases", async () => {
        const results = await amdResolve("foo/aliasButton/**/*", {
            targetExtension: "extA",
            cwd: extensionAPath,
            extensionPrefix: false,
            aliases: {
                "foo/alias": "core",
                ootch: "lib"
            }
        });

        expect(results).to.deep.equal([]);
    });
```

```
AssertionError: expected [ 'foo/aliasButton/a' ] to deeply equal []
      + expected - actual

      -[
      -  "foo/aliasButton/a"
      -]
      +[]
```

Reply:
Good catch,
the aliases are the `paths` from the amd configuration, so it can't be a regexp, nor be multiple 
It should be solved now.